tests/unit/test_cube.py

```diff
@@ -470,18 +471,183 @@ def and_it_accepts_a_JSON_format_cube_response(self, cube_response, expected_val
             ),
         ),
     )
-    def but_it_raises_on_other_cube_response_types(self, cube_response, expected_value):
+    def but_it_raises_on_other_cube_response_types(
+        self,
+        cube_response,
+        expected_value,
+    ):
         with pytest.raises(TypeError) as e:
             Cube(cube_response)._cube_dict
 
         assert str(e.value) == expected_value
 
+    @pytest.mark.parametrize(
+        "cube_response, expected_value",
```

Comment:
This is unreadable. I'd prefer if we tested this in a different way. I.e. if the same path is tested by integration tests, I'd say to get rid of this test.

If we would like to test if the dimensions are inflated properly, there's no need to replicate the entire `"measures"` objects in the response. We can just mock that with a property mock, and use a smaller piece of the cube response to test if the dimension is added. We don't even need to provide the cube with the response, but just test if the inflation is correctly executed (i.e. we could just test if a single element is being appended or inserted).